src/Microsoft.DotNet.Wpf/src/Shared/MS/Utility/BindUriHelper.cs

```diff
@@ -88,7 +87,7 @@ static internal Uri BaseUri
         static internal bool DoSchemeAndHostMatch(Uri first, Uri second)
         {
             // Check that both the scheme and the host match. 
-           return (SecurityHelper.AreStringTypesEqual(first.Scheme, second.Scheme) && first.Host.Equals(second.Host) == true);
+            return string.Equals(first.Scheme, second.Scheme, StringComparison.OrdinalIgnoreCase) && string.Equals(first.Host, second.Host);
```

Comment:
Could you revert the change of `String.Equals(String)` to `String.Equals(String, String)`. Before your change, if `first.Host` was null it would throw a NullReferenceException and with your change it would not throw. This is a change in behavior which IMO does not provide any value.

```suggestion
            return string.Equals(first.Scheme, second.Scheme, StringComparison.OrdinalIgnoreCase) && first.Host.Equals(second.Host);
```

Reply:
@ThomasGoulet73 How would I get `Host` from Uri to be null?

Reply:
I agree I don't think it is possible for `Uri.Host` to be null. I'm still not a fan of unrelated changes that **could** change the behavior, it makes it harder to review. I don't mind if you want to keep your PR as is.

Reply:
@ThomasGoulet73 No worries, I'm always glad for your reviews.

I'll keep the change as I've made it with checking this exact fact on purpose, as it imho signifies the meaning of the compare.

Reply:
@ThomasGoulet73  @h3xds1nz 
This code here : https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Private.Uri/src/System/Uri.cs#L343-L360 doesn't it indicate that `Host` can be `null` ?

Reply:
@dipeshmsft `UriInfo` != `Uri`, while `UriInfo.Host` is nullable because it can be, `Uri.Host` is not.

One of the benefits of NRT annotations! Anyways in the exact function you've linked, it goes right into [CreateHostString](https://source.dot.net/System.Private.Uri/R/f4f0ffc0f7c2a51d.html) that does the necessary fixes.

And as for the return; https://source.dot.net/#System.Private.Uri/System/UriExt.cs,941 -> https://source.dot.net/#System.Private.Uri/System/UriExt.cs,966 (You first call `true`, then `false`, second time it hops into `CreateHostString`)

Reply:
My bad, never mind it. Although there is a change in behaviour, it looks good to me.

Reply:
Thanks, happy to see every bit of `SecurityHelper` gone.